src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -244,6 +244,37 @@ protected static bool IdentifiersMatch(ISyntaxFactsService syntaxFacts, string n
             return symbolsMatch;
         }
 
+        internal static bool OriginalSymbolsMatch(
+            ISymbol searchSymbol,
+            ISymbol symbolToMatch,
+            Solution solution,
+            CancellationToken cancellationToken)
+        {
+            if (ReferenceEquals(searchSymbol, symbolToMatch))
+            {
+                return true;
+            }
+
+            if (searchSymbol == null || symbolToMatch == null)
+            {
+                return false;
+            }
+
+            Compilation searchSymbolCompilation = null;
+            if (!TryGetCompilation(searchSymbol, solution, ref searchSymbolCompilation, cancellationToken))
```

Comment:
Because `TryGetCompilation()` is using the `ref` as a cache.  The method looks like:

``` C#
if (compilation == null)
{
    compilation = //
}

return true;
```